File: app/src/main/java/com/southernbox/springscrollview/MainAdapter.java
package com.southernbox.springscrollview;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

/**
 * Created by SouthernBox on 2017/3/23 0007.
 * 主页列表适配器
 */

public class MainAdapter extends RecyclerView.Adapter {

    private Context mContext;
    private List<String> mList;

    public MainAdapter(Context context, List<String> mList) {
        this.mContext = context;
        this.mList = mList;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_list, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final RecyclerView.ViewHolder holder, int position) {
        ViewHolder viewHolder = (ViewHolder) holder;
        viewHolder.tvName.setText(mList.get(position));
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }

    private class ViewHolder extends RecyclerView.ViewHolder {

        TextView tvName;

        ViewHolder(View itemView) {
            super(itemView);
            tvName = (TextView) itemView.findViewById(R.id.tv_name);
        }
    }
}

File: kotlin/src/main/java/com/southernbox/springscrollview/MainAdapter.kt
package com.southernbox.springscrollview

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.synthetic.main.item_list.view.*


/**
 * Created by SouthernBox on 2017/6/19 0019.
 * 主页列表适配器
 */

internal class MainAdapter(private val mContext: Context,
                           private val mList: List<String>)
    : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)
            : RecyclerView.ViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.item_list, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val viewHolder = holder as ViewHolder
        viewHolder.tvName.text = mList[position]
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    private inner class ViewHolder internal constructor(itemView: View)
        : RecyclerView.ViewHolder(itemView) {
        internal var tvName: TextView = itemView.tv_name
    }
}